### DISCUSSION TITLE
Best approach to handle image source from base64?

### DISCUSSION THREAD

**Discussion by andrerom** [saucecontrol/PhotoSauce#127] [Q&A] [answered]
Missing some sort of high level user documentation (as opposed to api doc) with real world examples, specifically wondering what the best approach is to process Image from base64 source to file on disk with MagicScaler.

So far I have something along the lines of the following, any suggestions for better handling?
```
byte[] bytes = Convert.FromBase64String(image64);// image64 is base64 string after ";base64,"

// full size (intend to encode to webp+jxl as well, but for now just testing defaults)
MagicImageProcessor.ProcessImage(bytes, filePath, ProcessImageSettings.Default);

// thumbnail
MagicImageProcessor.ProcessImage(bytes, filePath, getThumbnailSettings());
```

**Comment by saucecontrol** [comment, marked as answer]
That looks right to me.  As long as the base64-encoded data contains a normal image, decoding it to a byte array and passing that in will work the same as if you had the image in a file or `Stream` of some kind.
